lib/formatters/getById.js

```diff
@@ -1,23 +1,25 @@
 'use strict';
 
-module.exports = function (rows) {
-  var tags;
+module.exports = function (chal) {
+  var tags = [];
 
   // if no challenge was found
-  if (rows.length === 0) {
-    return {};
+  if (chal.length === 0) {
+    return [];
   }
 
+  if (chal[0].tags_id) {
+    tags = chal.map(function (obj) {
+      return {
+        id: obj.tags_id,
+        name: obj.tags_name
+      };
+    });
+  }
 
-  tags = rows.map(function (obj) {
-    return {
-      id: obj.tags_id,
-      name: obj.tags_name
-    };
-  });
-  delete rows[0].tags_id;
-  delete rows[0].tags_name;
-  rows[0].tags = tags;
+  delete chal[0].tags_id;
+  delete chal[0].tags_name;
+  chal[0].tags = tags;
 
-  return rows[0];
+  return chal[0];
```

Comment:
The type of the return of the function is different depending if a challenge has been found or not:
-> not found, returns an empty array []
-> found returns an object {...}

It feels a bit strange, don't we want to return always an array? @mk4111 @JMurphyWeb any thoughts on this?